software/database/src/main/java/org/apache/brooklyn/entity/database/mysql/MySqlNodeEffectors.java

```diff
@@ -75,7 +75,7 @@ public Void call(ConfigBag parameters) {
             DynamicTasks.queue(
                     SshTasks.newSshExecTaskFactory(machine, 
                                     "cd "+entity().getAttribute(MySqlNode.RUN_DIR),
-                                    "sed -i'' -e 's@^\\(\\s*password\\s*=\\s*\\).*$@\\1" + newPass.replace("\\", "\\\\") + "@g' mymysql.cnf")
+                                    "sed -i.bak 's/\\(password.*\\)=.*/\\1= " + newPass.replace("\\", "\\\\") + "/' mymysql.cnf")
```

Comment:
You are right for the 2 mentioned cases @neykov, I didn't think this through.

I'll need to go back to the drawing board.